prime-router/src/main/kotlin/azure/Event.kt

```diff
@@ -96,6 +102,64 @@ abstract class Event(val eventAction: EventAction, val at: OffsetDateTime?) {
         fun parseQueueMessage(event: String): Event {
             // validate incoming queue message is in the expected format. This will error out with an
             //  IllegalStateException and message if it is not valid
+            try {
+                val reportEventQueueMessage =
+                    JacksonMapperUtilities.defaultMapper.readValue<ReportEventQueueMessage>(event)
```

Comment:
// plain JSON
val node: ObjectNode = defaultMapper.read(event)
val type = node.getString("type")
when(type) {
  is "Convert
}

Reply:
val decodeAttempt: ReportQueueMessage? = decodeMessage<ReportEventQueueMessage>(event)

fun decodeMessage<T>(event {
 try{
  // parse
catch () {
null
}

}